modem/arq_data_type_handler.py

```diff
@@ -0,0 +1,83 @@
+# File: arq_data_type_handler.py
+
+import structlog
+import lzma
+import gzip
+
+class ARQDataTypeHandler:
+    def __init__(self):
+        self.logger = structlog.get_logger(type(self).__name__)
+        self.handlers = {
+            "raw": {
+                'prepare': self.prepare_raw,
+                'handle': self.handle_raw
+            },
+            "raw_lzma": {
+                'prepare': self.prepare_raw_lzma,
+                'handle': self.handle_raw_lzma
+            },
+            "raw_gzip": {
+                'prepare': self.prepare_raw_gzip,
+                'handle': self.handle_raw_gzip
+            },
+            "p2pmsg_lzma": {
```

Comment:
I was thinking about having two layers of data. At this lower level I would only say if the arq session payload is freedata data and how to get to it (compression algorithm) OR data for another app that might use the freedata platform (other data types).
Then, at the freedata level, we could potentially have multiple types of messages (p2p, etc ..) and this would be indicated in the JSON structure inside every freedata payload.
Does this make sense to you or have we already mixed everything up and so it doesn't matter anymore?

I think we are mixing two different levels of data here.
An ARQ data session transfer resulted in the transfer of n bytes with no additional meaning. The new data type byte field in ARQ sessions allows us to say roughly what's in those bytes. I was thinking this could be used to say if it's

Reply:
I think the additional data layer would make the entire system more scalable. If we are doing this at arq level and therefore inside this data handler, it would be bloated fast I think. 
So one use case, one endpoint should be the direction in my opinion. 

Inside message module we then can define several types with json structure. Which shouldn't be too much byte overhead because of compression. For this we could use the existing formatter with encapsulation?